### PR TITLE
add http status code handling for client

### LINKED ISSUE #87: status code handling for awc client
It seems currently the client span won't set span status to error when the http request fails in some ways (5xx, 4xx, etc), so the resulting spans' status are always UNSET.  Do you think its reasonable to automatically set the span status according to the http status similar to what golang's contrib `net/http` do? (https://github.com/open-telemetry/opentelemetry-go-contrib/blob/6c86403/instrumentation/net/http/otelhttp/handler.go#L219)

The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#status) have this included so at least we have a standard to follow.

### PR DESCRIPTION
**Pull request by Frefreak**
fix #87

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -183,8 +183,27 @@ impl InstrumentedClientRequest {
     }
 }
 
+// convert http status code to span status following the rules described by the spec:
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#status
+fn convert_status(status: http::StatusCode) -> (StatusCode, String) {
+    match status.as_u16() {
+        100..=399 => {
+            (StatusCode::Unset, String::from(""))
```
Comment (jtescher) [inline src/client.rs:191]:
```suggestion
            (StatusCode::Unset, String::new())
```

File: src/client.rs [hunk 2]
```diff
@@ -183,8 +183,27 @@ impl InstrumentedClientRequest {
...
+        },
+        // since we are the client, we MUST treat 4xx as error
+        400..=599 => {
+            (StatusCode::Error, String::from(""))
```
Comment (jtescher) [inline src/client.rs:195]:
```suggestion
            (StatusCode::Error, String::new())
```

**Review by jtescher — APPROVED** [review]
Looks good, just a couple small nits

File: src/client.rs (hunk 1 above)
Comment (jtescher) [reply src/client.rs:191]: Actually these would probably be cleaner as `(StatusCode::Unset, None)` and have the caller do `msg.unwrap_or_default()`

**Comment by Frefreak** [comment]
updated

**Comment by jtescher** [comment]
looks like the latest actix release broke some things

**Comment by Frefreak** [comment]
Seems to be related to this: https://github.com/actions-rs/clippy-check/issues/2

quote from clippy-check's README.md:
> Due to token permissions, this Action WILL NOT be able to post clippy annotations for Pull Requests from the forked repositories.

**Pull request merged**